File: C#/sorting.cs
using System;

namespace Loops {
    public class App {
        static void Main (string[] args) {
            int[] ar1 = { 6, 5, 1, 4, 3, 2 },
                ar2 = { 6, 5, 1, 4, 3, 2 },
                ar3 = { 6, 5, 1, 4, 3, 2 };
            int size = ar1.Length;
            Console.WriteLine ("Bubble Sort");
            bubbleSort (ref ar1, ref size);
            Console.WriteLine ("\nInsertion Sort");
            insertionSort (ref ar2, ref size);
            Console.WriteLine ("\nSelection Sort");
            selectionSort (ref ar3, ref size);
        }

        static void valueSwap (ref int x1, ref int x2) {
            int temp = x1;
            x1 = x2;
            x2 = temp;
        }

        static void bubbleSort (ref int[] ar, ref int size) {
            Console.WriteLine (string.Join (", ", ar));
            for (int i = 0; i < size; i++)
                for (int j = 0; j < size - i - 1; j++)
                    if (ar[j] > ar[j + 1]) {
                        valueSwap (ref ar[j], ref ar[j + 1]);
                        Console.WriteLine (string.Join (", ", ar));
                    }
        }

        static void insertionSort (ref int[] ar, ref int size) {
            Console.WriteLine (string.Join (", ", ar));
            int j, value = 0;
            for (int i = 1; i < size; i++) {
                value = ar[i];
                j = i - 1;
                while (j > -1 && ar[j] > value) {
                    ar[j + 1] = ar[j];
                    j--;
                }
                ar[j + 1] = value;
                Console.WriteLine (string.Join (", ", ar));
            }
        }

        static void selectionSort (ref int[] ar, ref int size) {
            Console.WriteLine (string.Join (", ", ar));
            int small;
            for (int i = 0; i < size - 1; i++) {
                small = i;
                for (int j = i + 1; j < size; j++) {
                    if (ar[j] < ar[small])
                        small = j;
                }
                valueSwap (ref ar[small], ref ar[i]);
                Console.WriteLine (string.Join (", ", ar));
            }
        }
    }
}

File: Python/sorting.py
def value_swap(ar, pos1, pos2):
    temp = ar[pos1]
    ar[pos1] = ar[pos2]
    ar[pos2] = temp


def bubbleSort(ar, size):
    print(ar)
    for i in range(0, size - 1):
        for j in range(0, size - i - 1):
            if ar[j] > ar[j + 1]:
                value_swap(ar, j, j + 1)
                print(ar)


def insertionSort(ar, size):
    print(ar)
    for i in range(1, size):
        value = ar[i]
        j = i - 1
        while j > -1 and ar[j] > value:
            ar[j + 1] = ar[j]
            j -= 1
        ar[j + 1] = value
        print(ar)


def selectionSort(ar, size):
    print(ar)
    small = 0
    for i in range(0, size - 1):
        small = i
        for j in range(i + 1, size):
            if ar[j] < ar[small]:
                small = j
        value_swap(ar, small, i)
        print(ar)


ar1 = [6, 5, 1, 4, 3, 2]
ar2 = [6, 5, 1, 4, 3, 2]
ar3 = [6, 5, 1, 4, 3, 2]
print("Bubble Sort")
bubbleSort(ar1, len(ar1))
print("\nInsertion Sort")
insertionSort(ar2, len(ar2))
print("\nSelection Sort")
selectionSort(ar3, len(ar3))
